js/ui/text_box/text_box.js

```diff
@@ -84,10 +84,6 @@ var TextBox = TextEditor.inherit({
     },
 
     _toggleMaxLengthProp: function() {
-        if(this._isAndroidOrIE()) {
-            return;
-        }
-
         var maxLength = this.option("maxLength");
         var isMaskSpecified = !!this.option("mask");
```

Comment:
I think you can remove this line and update condition below. Use the `_getMaxLength` method to get the `maxLength`